diff --git a/contracts/0.8.25/vaults/VaultHub.sol b/contracts/0.8.25/vaults/VaultHub.sol
index d65a346..e8bc8b9 100644
--- a/contracts/0.8.25/vaults/VaultHub.sol
+++ b/contracts/0.8.25/vaults/VaultHub.sol
@@ -179,6 +179,15 @@ contract VaultHub is PausableUntilWithRoles {
     /// @notice no unsettled obligations allowed on settlement
     uint256 internal immutable NO_UNSETTLED_ALLOWED = 0;
 
+    // -----------------------------
+    //           MOCKING FOR WRAPPER
+    // -----------------------------
+    bool public reportIsAlwaysFresh;
+
+    function mock__setReportIsAlwaysFresh(bool _reportIsAlwaysFresh) external {
+        reportIsAlwaysFresh = _reportIsAlwaysFresh;
+    }
+
     // -----------------------------
     //           IMMUTABLES
     // -----------------------------
@@ -1177,6 +1186,10 @@ contract VaultHub is PausableUntilWithRoles {
     }
 
     function _isReportFresh(VaultRecord storage _record) internal view returns (bool) {
+        if (reportIsAlwaysFresh) {
+            return true;
+        }
+
         uint256 latestReportTimestamp = _lazyOracle().latestReportTimestamp();
         return
             // check if AccountingOracle brought fresh report
