diff --git a/contracts/0.8.25/vaults/LazyOracle.sol b/contracts/0.8.25/vaults/LazyOracle.sol
index 1e57295..1d08604 100644
--- a/contracts/0.8.25/vaults/LazyOracle.sol
+++ b/contracts/0.8.25/vaults/LazyOracle.sol
@@ -303,6 +303,27 @@ contract LazyOracle is ILazyOracle, AccessControlEnumerableUpgradeable {
         );
     }
 
+    function mock__updateVaultData(
+        address _vault,
+        uint256 _totalValue,
+        uint256 _cumulativeLidoFees,
+        uint256 _liabilityShares,
+        uint256 _slashingReserve
+    ) external {
+        int256 inOutDelta;
+        (_totalValue, inOutDelta) = _handleSanityChecks(_vault, _totalValue, _storage().vaultsDataRefSlot);
+
+        _vaultHub().applyVaultReport(
+            _vault,
+            _storage().vaultsDataTimestamp,
+            _totalValue,
+            inOutDelta,
+            _cumulativeLidoFees,
+            _liabilityShares,
+            _slashingReserve
+        );
+    }
+
     /// @notice handle sanity checks for the vault lazy report data
     /// @param _vault the address of the vault
     /// @param _totalValue the total value of the vault in refSlot
