diff --git a/contracts/0.8.25/vaults/LazyOracle.sol b/contracts/0.8.25/vaults/LazyOracle.sol
index 01da492..29eff22 100644
--- a/contracts/0.8.25/vaults/LazyOracle.sol
+++ b/contracts/0.8.25/vaults/LazyOracle.sol
@@ -325,6 +325,28 @@ contract LazyOracle is ILazyOracle, AccessControlEnumerableUpgradeable {
         );
     }
 
+    function mock__updateVaultData(
+        address _vault,
+        uint256 _totalValue,
+        uint256 _cumulativeLidoFees,
+        uint256 _liabilityShares,
+        uint256 _slashingReserve
+    ) external {
+        uint256 vaultsDataTimestamp = _storage().vaultsDataTimestamp;
+        int256 inOutDelta;
+        (_totalValue, inOutDelta) = _handleSanityChecks(_vault, _totalValue, _storage().vaultsDataRefSlot, vaultsDataTimestamp, _cumulativeLidoFees);
+
+        _vaultHub().applyVaultReport(
+            _vault,
+            vaultsDataTimestamp,
+            _totalValue,
+            inOutDelta,
+            _cumulativeLidoFees,
+            _liabilityShares,
+            _slashingReserve
+        );
+    }
+
     /// @notice removes the quarantine for the vault
     /// @param _vault the address of the vault
     function removeVaultQuarantine(address _vault) external {
