diff --git a/contracts/0.8.25/vaults/LazyOracle.sol b/contracts/0.8.25/vaults/LazyOracle.sol
index 8407d8c..4a90e74 100644
--- a/contracts/0.8.25/vaults/LazyOracle.sol
+++ b/contracts/0.8.25/vaults/LazyOracle.sol
@@ -336,6 +336,36 @@ contract LazyOracle is ILazyOracle, AccessControlEnumerableUpgradeable {
         );
     }
 
+    function mock__updateVaultData(
+        address _vault,
+        uint256 _totalValue,
+        uint256 _cumulativeLidoFees,
+        uint256 _liabilityShares,
+        uint256 _slashingReserve
+    ) external {
+        uint256 vaultsDataTimestamp = _storage().vaultsDataTimestamp;
+        (uint256 checkedTotalValue, int256 inOutDelta) = _handleSanityChecks(
+            _vault,
+            _totalValue,
+            _storage().vaultsDataRefSlot,
+            vaultsDataTimestamp,
+            _cumulativeLidoFees,
+            _liabilityShares,
+            _maxLiabilityShares
+        );
+
+        _vaultHub().applyVaultReport(
+            _vault,
+            vaultsDataTimestamp,
+            checkedTotalValue,
+            inOutDelta,
+            _cumulativeLidoFees,
+            _liabilityShares,
+            _maxLiabilityShares,
+            _slashingReserve
+        );
+    }
+
     /// @notice removes the quarantine for the vault
     /// @param _vault the address of the vault
     function removeVaultQuarantine(address _vault) external {
